code/modules/vehicles/cars/car.dm

```diff
@@ -1,18 +1,34 @@
 /obj/vehicle/sealed/car
 	layer = ABOVE_MOB_LAYER
 	anchored = TRUE
+	default_driver_move = FALSE
 	var/car_traits = NONE //Bitflag for special behavior such as kidnapping
 	var/engine_sound = 'sound/vehicles/carrev.ogg'
 	var/last_enginesound_time
 	var/engine_sound_length = 20 //Set this to the length of the engine sound
 	var/escape_time = 200 //Time it takes to break out of the car
 
+/obj/vehicle/sealed/car/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 2
 /obj/vehicle/sealed/car/generate_actions()
 	. = ..()
 	initialize_controller_action_type(/datum/action/vehicle/sealed/remove_key, VEHICLE_CONTROL_DRIVE)
 	if(car_traits & CAN_KIDNAP)
 		initialize_controller_action_type(/datum/action/vehicle/sealed/DumpKidnappedMobs, VEHICLE_CONTROL_DRIVE)
 
+/obj/vehicle/sealed/car/driver_move(mob/user, direction)
+	if(key_type && !is_key(inserted_key))
+		to_chat(user, "<span class='warning'>[src] has no key inserted!</span>")
+		return FALSE
```

Comment:
It's how `/obj/vehicle/ridden` does it even though the key functionality is inherited from `vehicle`. I figured it would function basically the same here.
From vehicle:
 ```
/obj/vehicle/proc/driver_move(mob/user, direction)
	if(key_type && !is_key(inserted_key))
		to_chat(user, "<span class='warning'>[src] has no key inserted!</span>")
		return FALSE
	if(!default_driver_move)
		return
vehicle_move(direction)
```

Reply:
thats kevinz being a dumb

Reply:
@Qustinnus is right. This related code should be changed to properly be able to support additional move restrictions without duplicating code.